mesonbuild/scripts/python_info.py

```diff
@@ -80,17 +80,53 @@ def links_against_libpython():
 
 variables = sysconfig.get_config_vars()
 variables.update({'base_prefix': getattr(sys, 'base_prefix', sys.prefix)})
+sys_root_variables = variables
 
 is_pypy = '__pypy__' in sys.builtin_module_names
 
+# Determine some information from the sys_root if it is given in the
+# environment. When not given, it is expected to be an empty string.
+sys_root_prefix = os.environ.get('MESON_SYS_ROOT_PREFIX')
+if sys_root_prefix:
+    import glob
+    try:
+        # We need to load the sys_root sysconfigdata from its stdlib directory.
+        # We assume that the same Python version as the build machine exists
+        # within the sys_root with the same stdlib directory, save for the
+        # prefix. If not, we fall back to the build machine.
+        stdlib = sysconfig.get_path('stdlib', vars={
+            'base': sys_root_prefix,
+            'installed_base': sys_root_prefix
+        })
+
+        # We cannot predict exactly what the sys_root sysconfigdata filename
+        # will be, but there should only be one, so glob to find it.
+        sysconfigdatas = glob.glob(os.path.join(stdlib, '_sysconfigdata*.py'))
+
+        # Only use the sys_root sysconfigdata if we find exactly one. More than
+        # one is odd, so it is safer to fall back to the build machine.
+        if len(sysconfigdatas) == 1:
+            with open(sysconfigdatas[0], encoding='utf-8') as f:
+                code = compile(f.read(), sysconfigdatas[0], 'exec')
+            data = {}
+            exec(code, data)
+            sys_root_variables = data.get('build_time_vars', variables)
+    except OSError:
```

Comment:
Why are you catching `OSError` here? I don't see anything in the try-except block that can raise `OSError`.

Reply:
In case there are issues reading the sysconfigdata file, such as permissions.

Reply:
I don't think silently ignoring permission errors provides a great user experience.

Reply:
I'm quite happy for it to blow up, but this is something it didn't do before, and I didn't want to cause any unexpected surprises for other users. It may do the wrong thing if it falls back to the build machine, but it may also be the right thing. If you really want me to remove this, then I will.